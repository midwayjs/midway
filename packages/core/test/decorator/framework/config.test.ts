import {
  AllConfig,
  Config,
  INJECT_CUSTOM_PROPERTY,
  MetadataManager
} from '../../../src';

class Test {
  @Config()
  hhh: any;

  @Config('bbb')
  bbb: any;

  @AllConfig()
  allConfig: any;
}

describe('/test/framework/config.test.ts', () => {
  it('config decorator should be ok', () => {
    expect(MetadataManager.getPropertiesWithMetadata(INJECT_CUSTOM_PROPERTY, Test)).toMatchSnapshot();
  });
});
