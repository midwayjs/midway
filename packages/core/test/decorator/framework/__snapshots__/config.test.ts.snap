// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/config.test.ts config decorator should be ok 1`] = `
{
  "allConfig": [
    {
      "key": "config",
      "metadata": {
        "identifier": "common:all_value_key",
      },
      "options": {
        "impl": true,
      },
      "propertyName": "allConfig",
    },
  ],
  "bbb": [
    {
      "key": "config",
      "metadata": {
        "identifier": "bbb",
      },
      "options": {
        "impl": true,
      },
      "propertyName": "bbb",
    },
  ],
  "hhh": [
    {
      "key": "config",
      "metadata": {
        "identifier": undefined,
      },
      "options": {
        "impl": true,
      },
      "propertyName": "hhh",
    },
  ],
}
`;
